lib/wallet/api/http/Cardano/Wallet/Api/Http/Shelley/Server.hs

```diff
@@ -2623,6 +2633,43 @@ constructTransaction api argGenChange knownPools poolStatus apiWalletId body = d
 
     walletId = getApiT apiWalletId
 
+    parseReferenceScript
+        :: ApiConstructTransactionData n
+        -> (SlotNo, SlotNo)
+        -> Either ErrConstructTx (Maybe (Script Cosigner))
+    parseReferenceScript tx validity = do
+        let mbRefScript = tx ^. #referencePolicyScriptTemplate
+        for mbRefScript $ \(ApiT refScript) -> do
+            guardWrongScriptTemplate refScript
+            guardOutsideValidityInterval validity refScript
+            Right refScript
+      where
+        guardWrongScriptTemplate
+            :: Script Cosigner -> Either ErrConstructTx ()
+        guardWrongScriptTemplate apiScript =
+            when (wrongMintingTemplate apiScript)
+                $ Left ErrConstructTxWrongMintingBurningTemplate
+          where
+            wrongMintingTemplate script =
+                isLeft (validateScriptOfTemplate RecommendedValidation script)
+                || countCosigners script /= (1 :: Int)
+                || existsNonZeroCosigner script
```

Comment:
Why are these conditions checked? According to the specification (swagger.yaml), any script template should be acceptable. 🤔

Reply:
script template allows more than one cosigner in general. But here we are using script for minting/burning in SHELLEY (not in shared style) so we need to make sure one cosigner is present.

Reply:
That's alright with me — all I'm asking for is that this fact is highlighted in the specification. 🤓📓 Could you amend the documents [mint-burn.md][1] and [swagger.yaml][2] to describe the allowed values for `reference_policy_script_template`?

(By definition, if the implementation is different from the specification, that's a bug.)

  [1]: https://github.com/cardano-foundation/cardano-wallet/blob/master/specifications/api/mint-burn.md
  [2]: https://github.com/cardano-foundation/cardano-wallet/blob/master/specifications/api/swagger.yaml

Reply:
ah, ok. done

Reply:
Thanks for adding a description of the restriction to [mint-burn.md](https://github.com/cardano-foundation/cardano-wallet/blob/master/specifications/api/mint-burn.md) — could you add it to [swagger.yaml](https://github.com/cardano-foundation/cardano-wallet/blob/master/specifications/api/swagger.yaml) as well?